lib/dotcom_web/controllers/trip_plan_controller.ex

```diff
@@ -520,7 +520,8 @@ defmodule DotcomWeb.TripPlanController do
       type: type
     } = Enum.find(legs, &(Leg.route_id(&1) == {:ok, id}))
 
-    custom_route = %Route{
+    %Route{
+      external_agency_name: if(type == "2", do: "Massport"),
```

Comment:
Confusingly, `type` doesn't come from the route type, it comes from an OTP field with the agency ID. That parsing's getting overhauled, anyway.